### PR TITLE
UUPS review

### PR DESCRIPTION
**Pull request by hellwolf**
- Additional test cases for
  - UUPSProxiable selfDestruct scenario
  - UUPSProxiable deadlock scenario
- UUPSProxiable to include deadlock prevention code. This is for to prevent from any possible mistakes of clumsy protocol upgrade operators.
- AgreementBase to hard code corresponding host address, this makes CFA/IDA logic contract not usable by other hosts.

Notes to nosy hackers, everything is okay, just additional test cases, nothing to see here.

### CODE REVIEW

File: packages/ethereum-contracts/contracts/upgradability/UUPSProxiable.sol [hunk 1]
```diff
@@ -37,9 +40,14 @@ abstract contract UUPSProxiable is Initializable {
             proxiableUUID() == UUPSProxiable(newAddress).proxiableUUID(),
             "UUPSProxiable: not compatible logic"
         );
+        require(
```
[the remark this answers is not in the document]
Comment (hellwolf) [reply packages/ethereum-contracts/contracts/upgradability/UUPSProxiable.sol:43]: Yes, one of the new test case was added for that:

    it("[SECURITY-DeadloopProxiable] proxy should not be a proxiable", async () => {

File: packages/ethereum-contracts/contracts/mocks/SuperfluidDesctructorMock.sol [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: AGPLv3
+// solhint-disable
+pragma solidity 0.7.6;
+
+contract SuperfluidDesctructorMock {
```
Comment (0xdavinchee) [inline packages/ethereum-contracts/contracts/mocks/SuperfluidDesctructorMock.sol:5]: not a huge deal but a small typo: `SuperfluidDesctructorMock` => `SuperfluidDestructorMock` same w/ file name

File: packages/ethereum-contracts/contracts/upgradability/UUPSProxiable.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (0xdavinchee) [reply packages/ethereum-contracts/contracts/upgradability/UUPSProxiable.sol:43]: gotcha!

File: packages/ethereum-contracts/contracts/mocks/SuperfluidDesctructorMock.sol (hunk 2 above)
Comment (hellwolf) [reply packages/ethereum-contracts/contracts/mocks/SuperfluidDesctructorMock.sol]: fixed

**Review by 0xdavinchee — APPROVED** [review]
this looks good

**Pull request merged**